lib/md055.js

```diff
@@ -0,0 +1,28 @@
+// @ts-check
+
+"use strict";
+
+const { addErrorContext } = require("../helpers");
+const { filterByTypes } = require("../helpers/micromark.cjs");
+
+const noPipeErrorMessage = "Table row is missing a leading or trailing pipe: ";
+
+module.exports = {
+    "names": [ "MD055", "table-missing-border" ],
+    "description": "Table is missing leading or trailing pipe character",
+    "tags": [ "table" ],
+    "function": function MD055(params, onError) {
+        const tables = filterByTypes(params.parsers.micromark.tokens, ["table"]);
+
+        for (const table of tables) {
+            const rows = table.children.flatMap((t) => t.children)
+                .filter((token) => token.type === "tableRow" || token.type === "tableDelimiterRow");
+
+            for (const row of rows) {
+                if (!row.text.startsWith("|") || !row.text.endsWith("|")) {
+                    addErrorContext(onError, row.startLine, noPipeErrorMessage + row.text);
```

Comment:
Please have a look at the use of `left` and `right` parameters to make sure truncation happens in the right place for long strings: <https://github.com/DavidAnson/markdownlint/blob/main/lib/md021.js>. And let's split this so it creates one error if the beginning is missing and another error if the end is missing. That'll make passing left/right easy.

AND that change sets up for the next step (which you should NOT do as part of this PR and I am happy to do if you do not want to) which is to add `fixInfo` so this rule can be automatically fixed. I think that will be straightforward much of the time, though maybe lining up the end pipe will be messy if someone tried to format their table.

Reply:
I split it. It'd be great if you can add the `fixInfo`.